ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java

```diff
@@ -88,11 +90,13 @@ protected void onCreate(Bundle savedInstanceState) {
     if (getReactNativeHost().getUseDeveloperSupport() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
       // Get permission to show redbox in dev builds.
       if (!Settings.canDrawOverlays(getContext())) {
-        needsOverlayPermission = true;
         Intent serviceIntent = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION, Uri.parse("package:" + getContext().getPackageName()));
         FLog.w(ReactConstants.TAG, REDBOX_PERMISSION_MESSAGE);
         Toast.makeText(getContext(), REDBOX_PERMISSION_MESSAGE, Toast.LENGTH_LONG).show();
-        ((Activity) getContext()).startActivityForResult(serviceIntent, REQUEST_OVERLAY_PERMISSION_CODE);
+        if (canHandleIntent(serviceIntent)) {
```

Comment:
we perhaps want to put the log and a toast from the lines above into this if block

Reply:
As I said, this is just a temporary workaround and the app will still crash when trying to display redbox alert. Do you think we can come up with a more bullet-proof solution for this?